apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.js

```diff
@@ -280,10 +284,35 @@ router.get(
 router.get(
   '/',
   asyncHandler(async (req, res) => {
-    const variant_id =
+    let variant_id =
       res.locals.assessment.type === 'Exam' || typeof req.query.variant_id !== 'string'
         ? null
         : req.query.variant_id;
+
+    const isAssessmentAvailable =
+      res.locals.assessment_instance.open && res.locals.authz_result.active;
+
+    if (variant_id === null && !isAssessmentAvailable) {
+      // We can't generate a new variant in this case, so we
+      // fetch and display the most recent non-broken variant.
+      // If no such variant exists, we tell the user that a new variant
+      // cannot be generated.
+      const last_variant_id = await sqldb.queryOptionalRow(
+        sql.select_last_variant_id,
+        { instance_question_id: res.locals.instance_question.id },
+        IdSchema,
+      );
+      if (last_variant_id == null) {
+        res.locals.no_variant_exists = true;
+        res.status(403).render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+        return;
+      }
```

Comment:
This will not log a page view. I think this is OK; just making sure that we've thought about this and decided that this is the desired behavior. cc @jonatanschroeder

Reply:
That's a good point. I don't see a problem either way, though current behaviour (if I'm not mistaken) also does not log a page view in this context.

Reply:
Correct, currently we don't, though I think that's strictly because the current behavior is to error out hard (that is, I think this isn't ever behavior we specifically thought about or designed).

Reply:
Indeed, I meant that logging it would add a new log entry which didn't exist before, so it's not something an instructor would expect to find before this change, and we don't have reports of this being needed before.

Reply:
We _could_ log the page view here, but I don't think it is necessary. It's not something that an instructor would need to pay attention to.

Reply:
I think we're all in agreement, no need to log here!